packages/babel-plugin-transform-template-literals/src/index.js

```diff
@@ -83,19 +83,27 @@ export default declare((api, options) => {
           callExpressionInput.push(t.arrayExpression(raws));
         }
 
-        const init = t.callExpression(helperId, callExpressionInput);
-        annotateAsPure(init);
-        init._compact = true;
-        scope.push({
-          id: templateObject,
-          init,
-          // This ensures that we don't fail if not using function expression helpers
-          _blockHoist: 1.9,
-        });
+        const callExpression = t.callExpression(helperId, callExpressionInput);
+        annotateAsPure(callExpression);
+        callExpression._compact = true;
+
+        const lazyLoad = t.logicalExpression(
+          "||",
+          t.cloneNode(templateObject),
+          callExpression,
+        );
+
+        const conditionalAssignment = t.assignmentExpression(
+          "=",
+          t.cloneNode(templateObject),
+          lazyLoad,
+        );
```

Comment:
Nit: we can optimize a little bit more by doing the assignment inside the logical expression:

```js
tag(_templateObject || (_templateObject = _taggedTemplateLiteral(...)))